frontend/src/Teacher/AllTasks.vue

```diff
@@ -0,0 +1,202 @@
+<script setup lang="ts">
+import type { Config, ConfigColumns, Api as DataTableObject } from 'datatables.net';
+import DataTablesCore from 'datatables.net-bs5';
+import DataTable from 'datatables.net-vue3';
+import { ref, watchEffect } from 'vue';
+import Loader from '../components/Loader.vue';
+import { getDate, getFromAPI } from '../utilities';
+
+DataTable.use(DataTablesCore);
+
+type Task = {
+  id: string;
+  title: string;
+  subject: string;
+  date: Date;
+};
+
+type RawTask = Omit<Task, 'date'> & {
+  date: string;
+};
+
+type SortValue = 'asc' | 'desc';
+
+/**
+ * Get tasks from API
+ * @param subject Subject abbreviation
+ * @param count Count of records
+ * @param start Offset of records
+ * @param sort Type of sorting
+ * @param search Search query
+ * @returns Tuple of [total count, tasks]
+ */
+const getTasks = async (
```

Comment:
Any reason why not just an async function? :)

Reply:
I personally prefer using arrow functions, no special reason here.

Reply:
Ok.